DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -57,6 +57,14 @@ class DMProcBuilder {
             foreach (DMASTProcStatement statement in block.Statements) {
                 try
                 {
+                    // ProcessStatementSet() needs to be before any loops
```

Comment:
This doesn't have any affect. It's just moving the `DMASTProcStatementSet` check outside of `ProcessStatement()`, it's still being processed in the same order.